watcher.go

```diff
@@ -200,6 +227,9 @@ func (w *Watcher) subscribe() error {
 	}
 	defer psc.Unsubscribe()
 
+	if w.options.RecordMetrics {
+		w.metricsCallback(w.createMetrics(subscribeMetric))
+	}
```

Comment:
Instrument Receive.  In the error case include the error.  In the Message case, record size of incoming message.

Reply:
@rwbarber2 done